src/docs/release/breaking-changes/rendereditable-stop-painting-invalid-selection.md

```diff
@@ -0,0 +1,96 @@
+---
+title: Change RenderEditable to stop painting the caret when its selection is invalid
+description: RenderEditable no longer paints the caret when its selection is (-1, -1).
+---
+
+## Summary
+
+When a text field's selection is `TextSelection.collapsed(offset: -1)`,
+`RenderEditable` no longer paints the caret, nor will its `onCaretChanged` 
+callback be called.
+
+## Context
+
+The caret indicates the insertion point within the current text in an 
+active input field. Typically, when a new character is entered, the 
+caret stays immediately after it. In Flutter the caret position is 
+represented by a collapsed selection. When the selection is invalid, 
+usually the user won't be able to modify or add text until they 
+change the selection to a valid value.
+
+The `RenderEditable` class is used to paint all types of text fields
+in Flutter today. Previously, `RenderEditable` paints the caret at 
+the start of the the document when the selection is invalid (range `(-1, -1)`),
+as if the selection was set to `(0, 0)`. This can be misleading as the 
+user would expect new input to be inserted at the start of the document, 
+which may not be the case for some input methods. The incorrect coordinates 
+of the caret are also reported by the `RenderEditable.onCaretChanged` 
+callback.
+
+## Description of change
+
+When `RenderEditable.selection` is set to `TextSelection.collapsed(offset: -1)`, 
+`RenderEditable.paint` no longer paints the caret, and the 
+`RenderEditable.onCaretChanged` callback is no longer called on this occasion.
+
+## Migration guide
+
+Common failures this change may introduce are golden test failures where:
+  - The caret is not shown when a text field's selection is invalid.
+  - The scroll offsest of a scrollable container is different (because text fields currently rely on coordinates reported by the`RenderEditable.onCaretChanged` callback to keep themselves on screen as the user types).
```

Comment:
If I hard break at 80 characters for this line, the preview text also has this line-break (but not the other ones).